data/data/ovirt/masters/main.tf

```diff
@@ -1,15 +1,26 @@
 resource "ovirt_vm" "master" {
-  count       = var.master_count
-  name        = "${var.cluster_id}-master-${count.index}"
-  cluster_id  = var.ovirt_cluster_id
-  template_id = var.ovirt_template_id
-  cores       = var.ovirt_master_cpu
-  memory      = var.ovirt_master_mem
+  count            = var.master_count
+  name             = "${var.cluster_id}-master-${count.index}"
+  cluster_id       = var.ovirt_cluster_id
+  template_id      = var.ovirt_template_id
+  instance_type_id = var.ovirt_master_instance_type_id
+  type             = var.ovirt_master_vm_type
+  cores            = var.ovirt_master_cores
+  sockets          = var.ovirt_master_sockets
+  // if instance type is declared then memory is redundant. Since terraform
+  // doesn't allow to condionally omit it, it must be passed.
+  // The number passed is multiplied by 4 and becomes the maximum memory the VM can have.
+  memory = var.ovirt_master_instance_type_id != "" ? 16348 : var.ovirt_master_memory
 
   initialization {
     host_name     = "${var.cluster_id}-master-${count.index}"
     custom_script = var.ignition_master
   }
+
+  block_device {
+    interface = "virtio_scsi"
+    size      = var.ovirt_master_os_disk_size_gb
+  }
```

Comment:
can you include why this change is part of this PR? in the commit message

Reply:
done

Reply:
this will extend the disk size if the machine spec definition is bigger than the definition in the VM template.
will add  to the commit in the commit message